### PR TITLE
add "tidy" feature

### PR DESCRIPTION
**Pull request by kevinb7**
I improved the usability since the demo with the following fixes:
- show .disable-overlay while the tidy-worker is working its magic
- overlay fades opacity in/out so people know it's there
- cursor position within a node is maintained so if it's in the middle of a literal it's there after tidying
- maintain relative position of active row so that it's in the same position after tidying
- preserve spacing between lines

For the last item I had to modify escodegen.  Right now package.json points to my branch.  I've submitted a pull request https://github.com/Constellation/escodegen/pull/208 for these changes.

### CODE REVIEW

**Comment by kevinb7** [comment]
I'm going to close this request for now.  I'll do a new pull request after the JSON dependency loading code (for web workers) is merged in.

**Comment by kevinb7** [comment]
I'll rebase this when I have some time.

File: build/workers/tidy/tidy-worker.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            }
+            if (line === loc.end.line && column > loc.end.column) {
+                return false;
+            }
+            return true;
+        } else {
+            return false;
+        }
+    };
+    walker.walk(ast, "program");
```
Comment (kevinb7) [inline build/workers/tidy/tidy-worker.js]: rewrite this using estraverse when rebasing

**Comment by kevinb7** [comment]
I started rebasing this, but it looks like cherry picking is going to be a lot easier.

**Comment by MatthiasSaihttam** [comment]
I really love the example you created for this, and would love to see this added to main webpage. It might work well inside the settings menu that we already have; all that's there now is size. Is there anything I can do get this included?

**Comment by kevinbarabash** [comment]
@MatthiasSaihttam yeah, I didn't think about putting it there.  Good suggestion.

**Pull request closed without merging**